vault/identity_store_entities.go

```diff
@@ -332,6 +332,11 @@ func (i *IdentityStore) pathEntityIDUpdate() framework.OperationFunc {
 			return logical.ErrorResponse("missing entity id"), nil
 		}
 
+		// Acquire the lock to modify the entity storage entry
+		lock := locksutil.LockForKey(i.entityLocks, entityID)
```

Comment:
I think you need to add this in pathEntityRegister too since it also calls handleEntityUpdateComon.

Reply:
The only case left in pathEntityRegister is new entities since when the id is provided it comes through this method.

Reply:
Yes, but you changed the function in handleEntityUpdateCommon to use a non-locking version, and now if it goes through pathEntityRegister and don't provide an ID that function is called without a lock.

Reply:
I added back the lock on a new entity by calling upsertEntity for new entities.